### DISCUSSION TITLE
Support for java 21

### DISCUSSION THREAD

**Discussion by joanbalaguero** [redis/lettuce#2470] [General]
Hello,

Is there any plan to support jdk21 for any of the next releases of lettuce? 

Also, any future plan to  support virtual threads for the sync connections?

Thanks,

Joan.

**Comment by mp911de** [comment]
What exactly are you looking for? Lettuce works fine on Java 21 and the differentiator to other libraries is that it uses an Event Loop for its I/O.

Switching to Virtual Threads as I/O event loop isn't possible really as Netty's threads are bound to the I/O implementation.

**Reply by joanbalaguero** [reply]
Hi Mark,

Ok perfect, that is what I was exactly looking for, just confirm lettuce-core runs fine on jdk21. We are migrating our api gateway from java11 to java21 and lettuce is one of the components we are using.

Thanks,

Joan.
